test/k23si/SKVClientTest.cpp

```diff
@@ -274,6 +276,585 @@ seastar::future<> runScenario03() {
+                                return seastar::make_ready_future<>();
+                            });
+                        })
+                        .then([&] {
+                            return txnHandle.read<k2::dto::SKVRecord>(std::move(read5))
+                            .then([](k2::ReadResult<k2::dto::SKVRecord>&& response) {
+                                K2EXPECT(response.status, k2::dto::K23SIStatus::OK);
+                                
+                                std::optional<k2::String> partkey = response.value.deserializeNext<k2::String>();
+                                std::optional<k2::String> rangekey = response.value.deserializeNext<k2::String>();
+                                std::optional<k2::String> data1 = response.value.deserializeNext<k2::String>();
+                                std::optional<k2::String> data2 = response.value.deserializeNext<k2::String>();
+                                K2EXPECT(*partkey, "partkey_s04");
+                                K2EXPECT(*rangekey, "rangekey_s04");
+                                K2EXPECT(*data1, "data1_v2"); // "f1" field was not updated successfully
+                                K2EXPECT(*data2, "data2_v3");
+                                
+                                return seastar::make_ready_future<>();
+                            });
+                        })
+                        .then([&] {
+                            // case 6
+                            return txnHandle.partialUpdate<k2::dto::SKVRecord>(rec6, {2,3})
+                            .then([](auto&& response) {
+                                K2EXPECT(response.status, k2::dto::K23SIStatus::Created);
+                                return seastar::make_ready_future<>();
+                            });
+                        })
+                        .then([&] {
+                            return txnHandle.read<k2::dto::SKVRecord>(std::move(read6))
+                            .then([](k2::ReadResult<k2::dto::SKVRecord>&& response) {
+                                K2EXPECT(response.status, k2::dto::K23SIStatus::OK);
+                                
+                                std::optional<k2::String> partkey = response.value.deserializeNext<k2::String>();
+                                std::optional<k2::String> rangekey = response.value.deserializeNext<k2::String>();
+                                std::optional<k2::String> data1 = response.value.deserializeNext<k2::String>();
+                                std::optional<k2::String> data2 = response.value.deserializeNext<k2::String>();
+                                K2EXPECT(*partkey, "partkey_s04");
+                                K2EXPECT(*rangekey, "rangekey_s04");
+                                K2EXPECT(*data1, "data1_v3");   // "f1" field was updated to v3 
+                                K2EXPECT(*data2, "");           // "f2" field was updated to null
```

Comment:
It would be better to use K2EXPECT(data2, false) in cases like this where you expect a null field.

Reply:
K2EXPECT(data2, false) cannot compile，while K2ASSERT(!data2, "msg") can. But assert doesn't look so pretty here.